src/main/kotlin/no/nav/helsearbeidsgiver/forespoersel/ForespoerselService.kt

```diff
@@ -1,7 +1,39 @@
 package no.nav.helsearbeidsgiver.forespoersel
 
+import no.nav.helsearbeidsgiver.utils.log.sikkerLogger
+
 class ForespoerselService(
     private val forespoerselRepository: ForespoerselRepository,
 ) {
-    fun hentForespoerslerForOrgnr(orgnr: String): List<Forespoersel> = forespoerselRepository.hentForespoerslerForOrgnr(orgnr)
+    fun hentForespoerslerForOrgnr(orgnr: String): ForespoerselResponse {
+        kotlin
+            .runCatching {
+                sikkerLogger().info("Henter forespørsler for bedrift: $orgnr")
+                forespoerselRepository.hentForespoerslerForOrgnr(orgnr)
+            }.onSuccess {
+                sikkerLogger().info("Hentet ${it.size} forespørsler for bedrift: $orgnr")
+                return ForespoerselResponse(it.size, it)
+            }.onFailure {
+                sikkerLogger().warn("Feil ved henting av forespørsler for bedrift: $orgnr", it)
+                throw it
+            }
+        throw RuntimeException("Feil ved henting av forespørsler for bedrift: $orgnr")
```

Comment:
Er denne nødvendig? Trodde det skulle holde å kaste i onFailure()
Tror også du kan kutte ut kotlin og bare skrive runCatching?

Reply:
hvis du tenker på RuntimeException så er det nødvendig. man må enten returnere noe eller kaste exception. den vil aldri bli kastet tror jeg men kompilatoren krever det.

Reply:
Ok, men du kan kutte ut throw it i .onFailure i så fall

Reply:
Da får vi ikke årsaken til feilen.

Reply:
nei, men den logges